src/modalities/nn/weight_init/weight_init.py

```diff
@@ -0,0 +1,56 @@
+from typing import List
+
+import torch.nn as nn
+
+from modalities.nn.weight_init.weight_init_if import WeightInitializationIF
+
+
+class WeightInitializerWrapper(WeightInitializationIF):
+    def __init__(self, weight_initializers: List[WeightInitializationIF]):
+        self.weight_initializers = weight_initializers
+
+    def initialize_in_place(self, model: nn.Module):
+        for weight_initializer in self.weight_initializers:
+            weight_initializer.initialize_in_place(model)
+
+
+class ModulewiseNormalInitialization(WeightInitializationIF):
+    def __init__(self, mean: float, std: float):
+        """Initializes the weights of a model by sampling from a normal distribution.
+        NOTE: This class supports the initialization of nn.Linear and nn.Embedding layers.
+        For other layer types, the initialization must be subclassed and extended
+
+        Args:
+            mean (float): mean of the normal distribution
+            std (float): standard deviation of the normal distribution
+        """
+        self.mean = mean
+        self.std = std
+
+    def _init_weights_impl(self, module: nn.Module):
+        if isinstance(module, nn.Linear):
+            nn.init.normal_(module.weight, mean=self.mean, std=self.std)
+            if module.bias is not None:
+                nn.init.zeros_(module.bias)
+        elif isinstance(module, nn.Embedding):
+            nn.init.normal_(module.weight, mean=self.mean, std=self.std)
+        else:
+            pass
+            # TODO need to check if we want raise an exception here
```

Comment:
TODO

Reply:
Models other than GPT-2 will have modules that are neither `nn.Linear` nor `nn.Embedding`. For instance, CoCa contains `nn.Conv2d`, see [here](https://github.com/Modalities/modalities/blob/bb4cd363a619792c8d00ccb78416893d0693b354/src/modalities/models/vision_transformer/vision_transformer_model.py#L40). However, `nn.Conv2d` also has the attributes `weight` and `bias` (see [here](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html)). So a pragmatic, first solution could be something like this: We check in the `else` part whether a module has attributes `weight` and (potentially) `bias`. If it does, we use the same initialization as for `nn.Linear`. If it contains other attributes, we raise an Exception.

Reply:
handled here: https://github.com/Modalities/modalities/pull/168/commits/48dced93e783090917fd3a9ba82046064f09d0a8

Reply:
It is implemented here now: 
https://github.com/Modalities/modalities/blob/b8e42c10dba4b61d97a06e21a13ad679ef6cc1cd/src/modalities/nn/weight_init/parameter_name_regex_filters.py#L71-L72